### PR TITLE
MM-39711 - Gekidou Settings Timezone

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary
This PR adds in the Timezone setting to the app.

I have left a todo in `display_timezone.tsx` file for when the user has multiple 'manual' timezones and chooses to use automatic one. I'm not sure what to do there.

#### Ticket Link 
https://mattermost.atlassian.net/browse/MM-39711

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/screens/settings/display_timezone_select/index.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        return timezones.filter((t) => (
+            getTimezoneRegion(t).toLowerCase().indexOf(lowerCasePrefix) >= 0 ||
+            t.toLowerCase().indexOf(lowerCasePrefix) >= 0
+        ));
+    };
+
+    const onPressTimezone = useCallback((tzne: string) => {
+        onBack(tzne);
+        popTopScreen();
+    }, []);
```
Comment (larkox) [inline app/screens/settings/display_timezone_select/index.tsx:80]: In theory, this depends on `onBack`.

**Review by larkox — APPROVED** [review]
LGTM, just one nit.

File: app/screens/settings/display_timezone/display_timezone.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {View} from 'react-native';
+import {Edge, SafeAreaView} from 'react-native-safe-area-context';
+
+import {updateMe} from '@actions/remote/user';
+import OptionItem from '@app/components/option_item';
```
Comment (enahum) [inline app/screens/settings/display_timezone/display_timezone.tsx:10]:
```suggestion
import OptionItem from '@components/option_item';
```

File: app/screens/settings/display_timezone/display_timezone.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+    const updateAutomaticTimezone = (useAutomaticTimezone: boolean) => {
+        const automaticTimezone = getDeviceTimezone();
+        setUserTimezone((prev) => ({
+            ...prev,
+            useAutomaticTimezone,
+            automaticTimezone,
+        }));
+
+        //todo: confirm with Elias
```
Comment (enahum) [inline app/screens/settings/display_timezone/display_timezone.tsx:72]: what do you need to confirm?

File: app/screens/settings/display_timezone/index.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+
+import {observeCurrentUser} from '@queries/servers/user';
+import {WithDatabaseArgs} from '@typings/database/database';
```
Comment (enahum) [inline app/screens/settings/display_timezone/index.tsx:8]:
```suggestion
import type {WithDatabaseArgs} from '@typings/database/database';
```

File: app/screens/settings/display_timezone_select/index.tsx (hunk 1 above)
Comment (enahum) [reply app/screens/settings/display_timezone_select/index.tsx:80]: ☝️

File: app/screens/settings/display_timezone_select/index.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                const allTzs = await client.getTimezones();
+                if (Array.isArray(allTzs)) {
+                    setTimezones(allTzs);
+                    const timezoneIndex = allTzs.findIndex((timezone) => timezone === selectedTimezone);
+                    if (timezoneIndex > 0) {
+                        setInitialScrollIndex(timezoneIndex);
+                    }
+                }
+            } catch (error) {
+                setTimezones(EMPTY_TIMEZONES);
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/settings/display_timezone_select/index.tsx:106]: ☝️ but I would want to log the error though

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/screens/settings/display_timezone/display_timezone.tsx (hunk 3 above)
Comment (avinashlng1080) [reply app/screens/settings/display_timezone/display_timezone.tsx:72]: Please disregard this comment.  I forgot to remove it.

**Comment by avinashlng1080** [comment]
I took the liberty of updating all the import statements in the app.

File: app/screens/settings/display_timezone_select/index.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+                    const timezoneIndex = allTzs.findIndex((timezone) => timezone === selectedTimezone);
+                    if (timezoneIndex > 0) {
+                        setInitialScrollIndex(timezoneIndex);
+                    }
+                }
+            } catch (error) {
+                // eslint-disable-next-line no-console
+                console.log('FAILED TO GET ALL TIMEZONES', error);
+            }
+        };
```
Comment (enahum) [inline app/screens/settings/display_timezone_select/index.tsx:109]: we should make this a remote action even if it does not store data in the db

File: app/screens/settings/display_timezone/display_timezone.tsx (hunk 3 above)
Comment (enahum) [reply app/screens/settings/display_timezone/display_timezone.tsx:72]: so what about this?

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/screens/settings/display_timezone_select/index.tsx (hunk 6 above)
Comment (avinashlng1080) [reply app/screens/settings/display_timezone_select/index.tsx]: corrected

**Review by enahum — APPROVED** [review]

**Pull request merged**